### PR TITLE
docs(SPEC): add syntax for type expressions

### PR DESCRIPTION
**Pull request by nathanielc**
### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -830,6 +830,86 @@ Examples:
...
+
+When an object's upper bound is not specified, it is assumed to be equal to its lower bound.
+
+Parameters to function types define whether the parameter is a pipe forward parameter and whether the parameter has a default value.
+The `<-` indicates the parameter is the pipe forward parameter.
+
+Examples:
+ 
+    // alias the bool type
+    type boolean = bool
```
Comment (jlapacik) [inline docs/SPEC.md]: Are named types aliases? To be more precise, is `boolean` the same type as `bool`? Can `boolean` be used wherever `bool` can and vice versa?

**Comment by nathanielc** [comment]
> Are named types purely for built-in declarations or can type expressions be used to explicitly annotate the type of any value defined in Flux?

Currently no, type expressions are only for built in statements. 

Type annotations may be added later but this change to the SPEC does not require it.

File: docs/SPEC.md (hunk 1 above)
Comment (nathanielc) [reply docs/SPEC.md]: Yes, they are equivalent in every way.

File: docs/SPEC.md [hunk 2]
```diff
@@ -830,6 +830,86 @@ Examples:
...
+        city: string,
+        state: string,
+        country: string,
+        province: string,
+        zip: int,
+    }
+
+### Types as values
+
+A special type named `type` describes values which can hold any other type.
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply docs/SPEC.md]: I may be over thinking this but the use case is reflection. 

For example say we create a function called `doc` that returns a string of the documentation for a value. Documentation is likely to be tied to a type, so being value to store a type as a value enables such a function.

```
doc = (v) => typeof(v:v).documents
```

Where `typeof` is a function that returns the type of a value.

[the remark this answers is not in the document]
Comment (nathanielc) [reply docs/SPEC.md]: That said I am going to remove it from the SPEC for now and we can add it later when we have a more concrete understanding.

**Pull request merged**